crawlers/mooncrawl/mooncrawl/leaderboards_generator/utils.py

```diff
@@ -101,3 +106,120 @@ def get_results_for_moonstream_query(
                 keep_going = num_retries <= max_retries
 
     return result
+
+
+def get_data_from_url(url):
+    response = requests.get(url)
+    if response.status_code == 200:
+        return response.json()
+    else:
+        raise Exception(f"Failed to get data: HTTP {response.status_code}")
+
+
+def chunk_data(data, chunk_size=100000):
+    for i in range(0, len(data), chunk_size):
+        yield data[i : i + chunk_size]
+
+
+def send_data_to_endpoint(chunks, endpoint_url, headers, timeout=10):
+    for index, chunk in enumerate(chunks):
+        try:
+            logger.info(f"Pushing chunk {index} to leaderboard API")
+            response = requests.put(
+                endpoint_url, headers=headers, json=chunk, timeout=timeout
+            )
+
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as http_error:
+            logger.error(
+                f"Could not push results to leaderboard API: {http_error.response.text} with status code {http_error.response.status_code}"
+            )
+            continue
+
+
+def leaderboard_push_batch(
+    leaderboard_id: str,
+    leaderboard_config: Dict[str, Any],
+    data: List[Dict[str, Any]],
+    headers: Dict[str, str],
+    batch_size: int = MOONSTREAM_LEADERBOARD_GENERATOR_BATCH_SIZE,
+    timeout: int = 10,
+) -> None:
+    """
+    Push leaderboard data to the leaderboard API in batches.
+    """
+
+    ## first step create leaderboard version
+
+    leaderboard_version_api_url = (
+        f"{MOONSTREAM_ENGINE_URL}/leaderboard/{leaderboard_id}/versions"
+    )
+
+    json_data = {
+        "publish": False,
+    }
+
+    leaderboard_api_response = requests.post(
+        leaderboard_version_api_url, json=json_data, headers=headers, timeout=5
+    )
+
+    try:
+        leaderboard_api_response.raise_for_status()
+    except requests.exceptions.HTTPError as http_error:
+        logger.error(
+            f"Could not create leaderboard version: {http_error.response.text} with status code {http_error.response.status_code}"
+        )
+        return
+
+    leaderboard_version_id = leaderboard_api_response.json()["version"]
+
+    ## second step push data to leaderboard version
+
+    leaderboard_version_push_api_url = f"{MOONSTREAM_ENGINE_URL}/leaderboard/{leaderboard_id}/versions/{leaderboard_version_id}/scores?normalize_addresses={leaderboard_config['normalize_addresses']}&overwrite=false"
+
+    chunks = chunk_data(data, chunk_size=batch_size)
```

Comment:
You can also use a generator comprehension here instead of defining `def chunk_data...` function:

```python
chunks = (data[i:i+batch_size] for i in range(0, len(data), batch_size))
```

See this for more details on generator comprehensions: https://python-reference.readthedocs.io/en/latest/docs/comprehensions/gen_expression.html